analyses/mutational-signatures/07-plot_cns_fit.Rmd

```diff
@@ -0,0 +1,364 @@
+  dplyr::select(cancer_group_hex, cancer_group_display) %>%
+  dplyr::distinct() %>%
+  dplyr::inner_join(exposure_counts) %>%
+  dplyr::mutate(
+    cancer_group_display_n = stringr::str_wrap(
+      glue::glue("{cancer_group_display} (N={total})"),
+      30)
+  ) %>%
+  dplyr::select(cancer_group_display_n, cancer_group_hex, total) %>%
+  dplyr::distinct() %>%
+  # factor cancer groups in order of number of samples
+  dplyr::mutate(cancer_group_display_n = forcats::fct_reorder(cancer_group_display_n, total, .desc=T)) 
+
+
+  
+# Grab named hex to incorporate into plot
+cancer_hex <- labels_hex$cancer_group_hex
+names(cancer_hex) <- labels_hex$cancer_group_display_n 
+
+
+exposures_sina_boxplot <- df_signature_mutations_per_mb %>%
+  # keep cancer_group_hex for joining
+  dplyr::select(cancer_group_hex, signature, exposure) %>%
+  # do NOT remove 0's!
+  #dplyr::filter(exposure > 0) %>%
+  dplyr::inner_join(labels_hex) %>%
+  ggplot2::ggplot() + 
+  ggplot2::aes(x = signature, 
+               y = exposure,
+               color = cancer_group_hex) + 
+  ggforce::geom_sina(size = 0.7) + 
+  ggplot2::geom_boxplot(outlier.size = 0, 
+                        size = 0.3, 
+                        color = "black", 
+                        alpha = 0,
+                        # remove wiskers
+                        coef = 0) +
+  ggplot2::facet_wrap(~cancer_group_display_n, nrow = 2) +
+  ggplot2::scale_color_identity() + 
+  ggplot2::scale_y_continuous(limits = c(0, 1.2), 
+                              breaks = c(0, 0.25, 0.5, 0.75, 1.0)) +
+  ggplot2::labs(
+    x = "RefSig signature",
+    y = "Signature weights across samples"
+  ) +
+  ggpubr::theme_pubr() + 
+  cowplot::panel_border() +
+  ggplot2::theme(
+    axis.text.x = ggplot2::element_text(size = 7.5),
+    axis.text.y = ggplot2::element_text(size = 10),
+    axis.title = ggplot2::element_text(size = 12),
+    strip.text = ggplot2::element_text(size = 8.5),
+    legend.position = "none"
+  )
+
+exposures_sina_boxplot
+
+ggplot2::ggsave(exposures_boxplot_file, exposures_sina_boxplot, width = 11, height = 4.5)
+  
+
+```
+
+<br><br>
+
+
+### Barplot of sample exposures faceted across cancer groups
+
+```{r barplot1, fig.width = 12, fig.height = 9}
+
+# Make a column for exposure of interest to order samples by
+samples_in_order <- df_signature_mutations_per_mb %>%
+  dplyr::filter(signature == 1) %>%
+  dplyr::select(exposure, Kids_First_Biospecimen_ID) %>%
+  dplyr::arrange(-exposure) %>%
+  dplyr::pull(Kids_First_Biospecimen_ID)
+
+
+sample_exposures_barplot <- df_signature_mutations_per_mb %>%
+  # keep cancer_group_hex for joining
+  dplyr::select(cancer_group_hex, signature, exposure, Kids_First_Biospecimen_ID) %>%
+  dplyr::inner_join(labels_hex) %>%
+  dplyr::mutate(Kids_First_Biospecimen_ID = factor(Kids_First_Biospecimen_ID, levels = samples_in_order)) %>%
+  ggplot2::ggplot() +
+  ggplot2::aes(x = Kids_First_Biospecimen_ID, 
+               y = exposure, 
```

Comment:
So would we expect the exposures to _mostly_ sum to 1 and when that is not the case, it is due to filtering out low confidence assignments (whatever the `deconstructSigs` equivalent is for that concept)?

Reply:
Not necessarily. There is potential for unknown weight across samples, which is also embedded in the RDS and can be explored. The function does not require that every mutation be assigned to a signature, but rather to determine the extent to which each signature is present. A quick look at this shows this distribution for unknown proportions across samples - 
```
 Min.   :0.000000  
 1st Qu.:0.005315  
 Median :0.042708  
 Mean   :0.047542  
 3rd Qu.:0.072105  
 Max.   :0.255552 
```

Reply:
Actually, something we might want to do is add "unknown" as a signature category to be explicit about this in the viz. What do you think?

Reply:
That makes sense to me!